test/media/media_source_engine_unit.js

```diff
@@ -649,6 +649,50 @@ describe('MediaSourceEngine', () => {
 
       expect(videoSourceBuffer.timestampOffset).toBe(0.50);
     });
+
+    it('calls abort before setting timestampOffset', async () => {
+      const delay = async () => {
+        await Util.shortDelay();
+        videoSourceBuffer.updateend();
+      };
+      const initObject = new Map();
+      initObject.set(ContentType.VIDEO, fakeVideoStream);
+
+      await mediaSourceEngine.init(
+          initObject, /* forceTransmuxTS= */ false, /* sequenceMode= */ true);
+
+      // First, mock the scenario where timestampOffset is set to help align
+      // text segments. In this case, SourceBuffer mode is still 'segments'.
+      let reference = dummyReference(0, 1000);
+      let appendVideo = mediaSourceEngine.appendBuffer(
+          ContentType.VIDEO, buffer, reference, /* hasClosedCaptions= */ false);
+      // Wait for the first appendBuffer(), in segments mode.
+      await delay();
+      // Next, wait for abort(), used to reset the parser state for a safe
+      // setting of timestampOffset.
+      await delay();
```

Comment:
Sorry for the delay, I misinterpreted the sequence of promises within this function and spent some time understanding it better.

You're right that abort() doesn't require the unit test to manually trigger updateend() on the SourceBuffer, since it fakes onUpdateEnd_() to resolve the abort() promise. I was trying to figure out, then, where the third `await delay();` is coming from, since only calling it twice results in a timeout failure.

So I think the sequence is:
1. For initial `append()` // for text segment alignment
2. For `remove()` // to clear buffer
3. For second `append()` // https://github.com/shaka-project/shaka-player/blob/8d3d556edaa817af686e1577f0a0bad92d0c74d4/lib/media/media_source_engine.js#L692-L697

Lastly, the `await appendVideo` is for the function-scoped `MediaSourceEngine#appendBuffer()` Promise.

I was thinking https://github.com/shaka-project/shaka-player/blob/8d3d556edaa817af686e1577f0a0bad92d0c74d4/lib/media/media_source_engine.js#L963-L965 was somehow still triggering an 'updateend' in reality, which caused me to go down a rabbit hole of looking through the W3C specs on when this event is supposed to trigger.